client/my-sites/plugins/controller.js

```diff
@@ -138,6 +140,44 @@ export function jetpackCanUpdate( context, next ) {
 	next();
 }
 
+function waitForState( context ) {
+	return new Promise( ( resolve ) => {
+		const unsubscribe = context.store.subscribe( () => {
+			const state = context.store.getState();
+
+			const siteId = getSelectedSiteId( state );
+			if ( ! siteId ) {
+				return;
+			}
+
+			const currentPlan = getCurrentPlan( state, siteId );
+			if ( ! currentPlan ) {
+				return;
+			}
+			unsubscribe();
+			resolve();
+		} );
+		// Trigger a `store.subscribe()` callback
+		context.store.dispatch( fetchSitePlans( getSelectedSiteId( context.store.getState() ) ) );
```

Comment:
TIL! This looks like a particularly indirect way of listening in to the state.

Reply:
TIL -> me too. :D I found this method in the Hosting Config controller.